src/context/AuthContextProvider.tsx

```diff
@@ -1,164 +1,109 @@
-import Keycloak, {KeycloakConfig, KeycloakInitOptions} from "keycloak-js";
-import React, {createContext, useEffect, useState} from "react";
-import axios from "axios";
-import {AuthContextValues} from "./AuthContextValues";
-import {retrieveToken} from "../common/auth";
-
-// const realm: string = process.env.REACT_APP_REALM_NAME ? process.env.REACT_APP_REALM_NAME : "";
-// const clientID: string = process.env.REACT_APP_CLIENT_ID ? process.env.REACT_APP_CLIENT_ID : "";
-// const url: string = process.env.REACT_APP_FEDERATED_CATALOGUE_API_URL ? process.env.REACT_APP_FEDERATED_CATALOGUE_API_URL : "";
-//
-// console.log("realm", realm);
-// console.log("clientID", clientID);
-// console.log("url", url);
-// console.log("EDGE", process.env.REACT_APP_EDGE_API_URI);
-
-// /**
-//  * KeycloakConfig configures the connection to the Keycloak server.
-//  */
-// const keycloakConfig: KeycloakConfig = {
-//   realm: realm,
-//   clientId: clientID,
-//   url: url,
-// };
+import Keycloak, { KeycloakConfig, KeycloakInitOptions } from "keycloak-js";
+import React, { createContext, useEffect, useState, useMemo } from "react";
 
 const keycloakConfig: KeycloakConfig = {
   realm: "gaia-x",
   clientId: "portal",
   url: "https://fc-keycloak.gxfs.gx4fm.org/",
 };
 
-/**
- * KeycloakInitOptions configures the Keycloak client.
- */
+const keycloak = new Keycloak(keycloakConfig);
+
 const keycloakInitOptions: KeycloakInitOptions = {
-  // Configure that Keycloak will check if a user is already authenticated (when opening the app or reloading the page). If not authenticated the user will be send to the login form. If already authenticated the webapp will open.
-  onLoad: "login-required",
+  onLoad: "check-sso",
   checkLoginIframe: false,
-  pkceMethod: 'S256'
+  pkceMethod: "S256",
 };
 
-// Create the Keycloak client instance
-const keycloak = new Keycloak(keycloakConfig);
+interface AuthContextType {
+  isAuthenticated: boolean;
+  token: string;
+  login: () => Promise<void>;
+  logout: () => Promise<void>;
+  hasRole: (role: string) => boolean;
```

Comment:
How about to define an enum with the possible roles?

Reply:
After further discussions, the UserRoles will be introduced in further versions of the project.